lib/internal/webstreams/writablestream.js

```diff
@@ -581,13 +505,85 @@ ObjectDefineProperties(WritableStreamDefaultController.prototype, {
   [SymbolToStringTag]: getNonWritablePropertyDescriptor(WritableStreamDefaultController.name),
 });
 
+function InternalWritableStream(start, write, close, abort, highWaterMark, size) {
+  markTransferMode(this, false, true);
+  this[kType] = 'WritableStream';
+  this[kState] = createWritableStreamState();
+  this[kIsClosedPromise] = createDeferredPromise();
+
+  const controller = new WritableStreamDefaultController(kSkipThrow);
+  setupWritableStreamDefaultController(
+    this,
+    controller,
+    start,
+    write,
+    close,
+    abort,
+    highWaterMark,
+    size
+  )
+}
+
+ObjectSetPrototypeOf(InternalWritableStream.prototype, WritableStream.prototype);
+ObjectSetPrototypeOf(InternalWritableStream, WritableStream);
+
+function createWritableStream(start, write, close, abort, highWaterMark, size) {
+  const stream = new InternalWritableStream(start, write, close, abort, highWaterMark, size);
+
+  // For spec compliance the InternalWritableStream must be a WritableStream
+  stream.constructor = WritableStream;
+  return stream;
+}
+
 const isWritableStream =
   isBrandCheck('WritableStream');
 const isWritableStreamDefaultWriter =
   isBrandCheck('WritableStreamDefaultWriter');
 const isWritableStreamDefaultController =
   isBrandCheck('WritableStreamDefaultController');
 
+function createWritableStreamState() {
+  return {
+    close: createDeferredPromise(),
+    closeRequest: {
```

Comment:
```suggestion
  return {
    __proto__: null,
    close: createDeferredPromise(),
    closeRequest: {
      __proto__: null,
```

Reply:
I suppose we should also do this for `createDeferredPromise()` then? https://github.com/nodejs/node/blob/62ca05017d758e254eda2dfaf9678d2038a9235b/lib/internal/util.js#L537-L546

Reply:
Nah, because that function can be replaced with `PromiseWithResolvers` once the primordial lands, which isn't a null object.